csharp/src/Ice/OutgoingResponseFrame.cs

```diff
@@ -9,83 +9,92 @@
+        public int Size { get; private set; }
 
-        private readonly Encoding _payloadEncoding; // TODO: move to OutputStream
+        // Contents of the Frame
+        internal List<ArraySegment<byte>> Data { get; private set; }
+
+        // Position of the end of the payload, for Ice1 this is always the frame end.
+        private OutputStream.Position? _payloadEnd;
 
         /// <summary>Creates a new outgoing request frame with an OK reply status and a void return value.</summary>
-        /// <param name="current">The current parameter holds decoded header data and other information about the
-        /// request for which this method creates a response.</param>
-        public static OutgoingResponseFrame Empty(Current current)
-            => new OutgoingResponseFrame(current, ReplyStatus.OK, ArraySegment<byte>.Empty);
+        /// <param name="encoding">The encoding for the frame payload.</param>
+        public static OutgoingResponseFrame Empty(Encoding encoding)
+            => new OutgoingResponseFrame(encoding, ArraySegment<byte>.Empty);
 
         /// <summary>Creates a new outgoing response frame. It's a partial frame, and its payload needs to be
-        /// written later on with StartReturnValue/EndReturnValue.</summary>
-        /// <param name="current">The current parameter holds decoded header data and other information about the
-        /// request for which this method creates a response.</param>
-        public OutgoingResponseFrame(Current current) : base(current.Adapter.Communicator, Ice1Definitions.Encoding)
+        /// written later on with WritePayload.</summary>
+        /// <param name="encoding">The encoding for the frame payload.</param>
+        public OutgoingResponseFrame(Encoding encoding)
         {
-            RequestId = current.RequestId;
-            _payloadEncoding = current.Encoding;
-
-            WriteSpan(Ice1Definitions.ReplyHeader.AsSpan());
-            WriteInt(RequestId);
+            Encoding = encoding;
+            Encoding.CheckSupported();
+            Data = new List<ArraySegment<byte>>();
         }
 
         /// <summary>Creates a new outgoing response frame with the given reply status and payload.</summary>
-        /// <param name="current">The current parameter holds decoded header data and other information about the
-        /// request for which this method creates a response.</param>
-        /// <param name="replyStatus">The reply status.</param>
+        /// <param name="encoding">The encoding for the frame payload.</param>
         /// <param name="payload">The payload for this response frame.</param>
         // TODO: add parameter such as "bool assumeOwnership" once we add memory pooling.
-        public OutgoingResponseFrame(Current current, ReplyStatus replyStatus, ArraySegment<byte> payload)
-            : this(current)
+        public OutgoingResponseFrame(Encoding encoding, ArraySegment<byte> payload)
+            : this(encoding)
```

Comment:
Here is simpler to not create the OutputStream because the empty encapsulation always fit in the segment that is just created, for the request frame that is not true and an empty encapsulation can be write across segments.